examples/git/git_example.go

```
@@ -0,0 +1,162 @@
+		p.Run(distributed.Option())
+	} else if *isDockerCluster {
+		p.Run(distributed.Option().SetMaster("master:45326"))
+	} else {
+		p.Run()
+	}
+}
+
+func flipKey(newKeyIdx int) gio.Mapper {
+	return func(x []interface{}) error {
+		newKey := make([]interface{}, 1)
+		newKey[0] = x[newKeyIdx]
+		row := x[:newKeyIdx]
+		if len(x) > newKeyIdx+1 {
+			row = append(row, x[newKeyIdx+1:]...)
+		}
+		row = append(newKey, row...)
+		gio.Emit(row...)
+		return nil
+	}
+}
+
+//TODO: Update to new index approach
+func readBlob(x []interface{}) error {
+	repoPath := gio.ToString(x[1])
+	blobHash := plumbing.NewHash(gio.ToString(x[5]))
+	contents := []byte("")
+
+	if !blobHash.IsZero() {
+		r, err := gogit.PlainOpen(repoPath)
+		if err != nil {
+			return err
+		}
+
+		blob, err := r.BlobObject(blobHash)
+		if err != nil {
+			return err
+		}
+
+		reader, err := blob.Reader()
+		if err != nil {
+			return err
+		}
+
+		contents, err = ioutil.ReadAll(reader)
+		if err != nil {
+			return err
+		}
+	}
+
+	gio.Emit(x[0], x[1], x[2], x[3], x[4], x[5], contents)
+	return nil
+}
+
+func classifyLanguage(fileNameIdx int, contentIdx int) gio.Mapper {
+	return func(x []interface{}) error {
+		filename := gio.ToString(x[fileNameIdx])
+		content := x[contentIdx].([]byte)
+		lang := enry.GetLanguage(filename, content)
+		gio.Emit(append(x, lang)...)
+		return nil
+	}
+}
+
+//TODO: Update to new index approach
+func extractUAST(x []interface{}) error {
+	client, err := bblfsh.NewClient("0.0.0.0:9432")
+	if err != nil {
+		panic(err)
+	}
+
+	blob := gio.ToString(x[4])
+
+	res, err := client.NewParseRequest().Language("python").Content(blob).Do()
+	if err != nil {
+		panic(err)
+	}
+
+	if res.Response.Status == protocol.Fatal {
+		res.Language = ""
+	}
+
+	gio.Emit(x[0], x[1], x[2], x[3], x[4], x[5], res.Language)
+	return nil
+}
+
+func toPrint(val interface{}) string {
+	return fmt.Sprintf("%v", val)
+}
+
+func truncateString(str string, num int) string {
+	b := str
+	if len(str) > num {
```

do it the other way around to return early when possible

```go
func ellipsis(s string, n int) string {
	if len(s) <= n {
		return s
	}
	return s[:n-3] + "..."
}
```

if `n` is less than `3` I'd let it panic, as the call doesn't make sense